kratos/containers/nodal_data.h

```diff
@@ -0,0 +1,226 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Pooyan Dadvand
+//
+
+#if !defined(KRATOS_NODAL_DATA_H_INCLUDED )
+#define  KRATOS_NODAL_DATA_H_INCLUDED
+
+// System includes
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "containers/variables_list_data_value_container.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Stores all data and dofs which are stored in each elements
+/** This class is the container for nodal data storing:
+ *  Id : The Id of the node 
+*/
+class NodalData
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of NodalData
+    KRATOS_CLASS_POINTER_DEFINITION(NodalData);
+
+    using IndexType = std::size_t;
+
+    using SizeType = std::size_t;
+
+    using SolutionStepsNodalDataContainerType = VariablesListDataValueContainer;
+
+    using BlockType=VariablesListDataValueContainer::BlockType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    NodalData(IndexType TheId);
+
+    NodalData(IndexType TheId, VariablesList*  pVariablesList, BlockType const * ThisData, SizeType NewQueueSize = 1);
+
+    /// Destructor.
+    ~NodalData(){}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    NodalData& operator=(NodalData const& rOther);
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    IndexType Id() const
```

Comment:
Do you want me to add some Doxygen here?

Reply:
Added!